addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -29,6 +29,7 @@ class ChannelMember(models.Model):
     custom_channel_name = fields.Char('Custom channel name')
     fetched_message_id = fields.Many2one('mail.message', string='Last Fetched', index="btree_not_null")
     seen_message_id = fields.Many2one('mail.message', string='Last Seen', index="btree_not_null")
+    new_message_separator = fields.Many2one('mail.message', string='New Messages Separator Position', default=lambda s: s.seen_message_id, index="btree_not_null")
```

Comment:
Similarly, all the fields about these features should just be integer and not relations. This is just an id to say where is the separator, not an actual message reference that we want to keep.

If the message gets removed from DB, I'd expect to keep the reference where it was. What matters is the list/number of messages that are after the id.